camel/o1datagen/o1datagen.py

```diff
@@ -0,0 +1,365 @@
+        return answer
+
+    def verify_answer(self, question: str, answer: str) -> bool:
+        r"""Verify if a generated answer is semantically equivalent to
+            the golden answer for a given question.
+
+        Args:
+            question (str): The question to look up in the golden answers
+                dictionary.
+            answer (str): The answer to verify, typically generated
+                by a model or provided by a user.
+
+        Returns:
+            bool: True if the answer matches the golden answer based on
+                semantic equivalence (meaning the core content and meaning are
+                the same, even if the exact wording differs).
+                False in the following cases:
+                - If the provided question doesn't exist in the golden answers
+                - If the answer's meaning differs from the golden answer
+        """
+        golden_answer = self.golden_answers.get(question)
+        if golden_answer is None:
+            logger.error(f"Question '{question}' not found in golden answers.")
+            return False
+        prompt = f"""Please determine if the following two answers 
+        express the same meaning:
+        Question: {question}
+        Answer 1: {answer}
+        Answer 2: {golden_answer}
+        Just answer "True" or "False".
+        """
+        self.chat_agent.reset()
+        response = self.chat_agent.step(prompt)
+        is_correct = response.msgs[0].content.strip().lower() == "true"
+        logger.info("Answer verification result: %s", is_correct)
+        return is_correct
+
+    def monte_carlo_tree_search(
+        self, question: str, partial_solution: str = ""
+    ) -> tuple[str, bool]:
+        r"""Generate and verify answers using Monte Carlo Tree Search.
+
+        This method implements a Monte Carlo Tree Search approach
+        to find the best solution by iteratively generating answers
+        and scoring them against the golden answer.
+
+        Args:
+            question (str): The problem or question to be solved.
+            partial_solution (str, optional): A partial solution to build
+                upon. This canbe used to guide the search process with
+                existing progress. (default::obj:`""`)
+
+        Returns:
+            tuple[str, bool]: A tuple containing:
+                - str: The best solution found
+                (or empty string if no solution found)
+                - bool: True if a correct solution was found, False otherwise
+        """
+        logger.info("Starting Monte Carlo Tree Search")
+        best_solution = ""  # Initialize as empty string instead of None
+        best_score: float = 0.0
+        for i in range(self.search_limit):
+            # Generate new answer
+            current_solution = self.get_answer(question, partial_solution)
+            # Verify answer
+            is_correct = self.verify_answer(question, current_solution)
```

Comment:
currently `self.get_answer` and `self.verify_answer` is using the same `chat_agent`, should we set 2 different agent for different purpose? So that we can set different sys prompt and model for different target

Reply:
done it